### PR TITLE
[Feature][Connector-V2][Notion]Add Notion source connector

### PR DESCRIPTION
**Pull request by TaoZex**
## Purpose of this pull request

https://github.com/apache/incubator-seatunnel/issues/3018
Add Notion source connector

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

File: seatunnel-connectors-v2/connector-http/connector-http-notion/src/main/java/org/apache/seatunnel/connectors/seatunnel/notion/source/NotionSource.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    @Override
+    public String getPluginName() {
+        return "Notion";
+    }
+
+    @Override
+    public void prepare(Config pluginConfig) throws PrepareFailException {
+        CheckResult result = CheckConfigUtil.checkAllExists(pluginConfig, NotionSourceConfig.URL.key(), NotionSourceConfig.PASSWORD.key(), NotionSourceConfig.VERSION.key());
+        if (!result.isSuccess()) {
+            throw new PrepareFailException(getPluginName(), PluginType.SOURCE, result.getMsg());
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-http/connector-http-notion/src/main/java/org/apache/seatunnel/connectors/seatunnel/notion/source/NotionSource.java]: Please refer to https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-connectors-v2/connector-cassandra/src/main/java/org/apache/seatunnel/connectors/seatunnel/cassandra/source/CassandraSource.java#L65 and unified connector exception.

Comment (TaoZex) [reply seatunnel-connectors-v2/connector-http/connector-http-notion/src/main/java/org/apache/seatunnel/connectors/seatunnel/notion/source/NotionSource.java]: Thanks,I have fixed it.

**Comment by TaoZex** [comment]
@hailin0 @Hisoka-X @EricJoy2048 PTAL

File: seatunnel-connectors-v2/connector-http/connector-http-notion/src/main/java/org/apache/seatunnel/connectors/seatunnel/notion/source/NotionSourceFactory.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+public class NotionSourceFactory implements TableSourceFactory {
+    @Override
+    public String factoryIdentifier() {
+        return "Notion";
+    }
+
+    @Override
+    public OptionRule optionRule() {
+        return OptionRule.builder()
+                .required(NotionSourceConfig.URL)
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-http/connector-http-notion/src/main/java/org/apache/seatunnel/connectors/seatunnel/notion/source/NotionSourceFactory.java]: Lost `version` option?

Comment (TaoZex) [reply seatunnel-connectors-v2/connector-http/connector-http-notion/src/main/java/org/apache/seatunnel/connectors/seatunnel/notion/source/NotionSourceFactory.java]: Thanks.I have fixed it.

**Comment by EricJoy2048** [comment]
Need us to wait https://github.com/apache/incubator-seatunnel/issues/3573 finish ？

File: seatunnel-connectors-v2/connector-http/connector-http-notion/src/main/java/org/apache/seatunnel/connectors/seatunnel/notion/source/NotionSourceFactory.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+@AutoService(Factory.class)
+public class NotionSourceFactory implements TableSourceFactory {
+    @Override
+    public String factoryIdentifier() {
+        return "Notion";
+    }
+
+    @Override
+    public OptionRule optionRule() {
+        return OptionRule.builder()
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-http/connector-http-notion/src/main/java/org/apache/seatunnel/connectors/seatunnel/notion/source/NotionSourceFactory.java]: Add UT for optionRule() please.

Comment (TaoZex) [reply seatunnel-connectors-v2/connector-http/connector-http-notion/src/main/java/org/apache/seatunnel/connectors/seatunnel/notion/source/NotionSourceFactory.java]: Thanks. I have updated code. Please review it again.

**Review by EricJoy2048 — APPROVED** [review]
LGTM, Please waiting the CI complete.

**Comment by TaoZex** [comment]
> LGTM, Please waiting the CI complete.

Thanks.

**Review by TyrantLucifer — APPROVED** [review]

**Pull request merged**